clang/test/SemaSYCL/sycl-restrict.cpp

```diff
@@ -143,6 +143,9 @@ void usage(myFuncDef functionPtr) {
 
   // expected-error@+1 {{__float128 is not supported on this target}}
   __float128 A;
+
```

Comment:
add a bunch more tests.  I would very much like us to have the test coverage for the 'broken' ones you noticed as well, with a 'FIXME' or something in the text marking that it should be caught so that when the diagnostics code is reapplied it would work.

Reply:
added them here: clang/test/SemaSYCL/deferred-diagnostics-emit.cpp